runtime/lua/vim/diagnostic.lua

```diff
@@ -895,22 +895,30 @@ M.handlers.signs = {
     end
 
     local ns = M.get_namespace(namespace)
-    if not ns.user_data.sign_group then
-      ns.user_data.sign_group = string.format('vim.diagnostic.%s', ns.name)
+    if not ns.user_data.sign_ns then
+      ns.user_data.sign_ns = vim.api.nvim_create_namespace('')
     end
 
-    local sign_group = ns.user_data.sign_group
+    local sign_ns = ns.user_data.sign_ns
     for _, diagnostic in ipairs(diagnostics) do
-      vim.fn.sign_place(0, sign_group, sign_highlight_map[diagnostic.severity], bufnr, {
+      local lnum = diagnostic.lnum
+      local col = diagnostic.col
+      local sign_text = opts.sign_text and opts.sign_text[diagnostic.severity] or M.severity[diagnostic.severity]:sub(1,1)
```

Comment:
`sign_text` is a new option, so be sure to document it.

Reply:
I believe the docs are generated from the comments in the source via Doxygen correct? Is there a formatter for the comments?

Also, I think it would be better to change sign_text to a function `M.resolve_sign` which takes a diagnostic as its parameter, is public, and can be configured `via vim.diagnostic.config`, similar to what is implemented for the formatting of the virtual_text handler to allow for more advanced configurations.

Reply:
Adding a `format` key for the `signs` option would be useful (but can be done in a followup PR). One particular use case would be to always only show a single sign even when multiple sign columns are used (e.g., one for gitsigns and one for diagnostic signs).

Reply:
> Adding a `format` key for the `signs` option would be useful (but can be done in a followup PR). One particular use case would be to always only show a single sign even when multiple sign columns are used (e.g., one for gitsigns and one for diagnostic signs).

My experience has been that vim signs always take precedence over extmark signs (which is the bug that made me want to write this PR). I usually just have one sign appear, giving priority to diagnostics over gitsigns, but with these changes gitsigns has to use the extmark api. Gitsigns does have a config option for this, and everything seems to work smoothly when I set it, but it's a borderline breaking change since behavior will be really different for most people unless the author of gitsigns can change the defaults whenever this is merged.

Reply:
https://github.com/neovim/neovim/pull/19737

Reply:
(and the priority issue is fixed on master)

Reply:
> (and the priority issue is fixed on master)

Was that today? I ran into that issue while working 30 minutes ago and my branch was built from source yesterday I think. I

> (and the priority issue is fixed on master)

Oh that's perfect, I think that got pushed literally the day I opened this PR, that's some insane timing. I just rebased onto master so hopefully issue will be gone now.

Reply:
Not a coincidence (see the other comment thread below).